### PR TITLE
Sync sensor readings creation

### PR DESCRIPTION
**Pull request by haneslinger**
#### Any background context you want to provide?
 This syncs sensor readings creation to speed it up and gives better error messages.
#### How should this be manually tested?
try uploading `seed/tests/data/small_sensor_readings_example_bad.csv` to see erorrs.
try uploading `seed/tests/data/sensor_readings_example.csv`to see progress bar and speed.

### CODE REVIEW

File: seed/data_importer/tasks.py [hunk 1]
```diff
@@ -892,32 +892,69 @@ def _save_sensor_readings_data_create_tasks(file_pk, progress_key):
...
+    else:
+        try:
+            with transaction.atomic():
+                reading_strings = [
+                    f"({sensor.id}, '{timestamp}', '{value}')"
+                    for timestamp, value
+                    in readings_tuples
+                ]
+                sql = (
+                    'INSERT INTO seed_sensorreading(sensor_id, timestamp, reading)' +
```
Comment (nllong) [inline seed/data_importer/tasks.py:928]: very nice!

File: seed/static/seed/js/controllers/sensor_readings_upload_modal_controller.js [hunk 2]
```diff
@@ -113,9 +110,16 @@ angular.module('BE.seed.controller.sensor_readings_upload_modal', [])
...
+            {
+              field: 'exists',
+              enableHiding: false
+            }
+          ]
+
           $scope.proposed_imports_options = {
             data: result,
-            columnDefs: base_sensor_readings_col_defs,
+            columnDefs: [...base_sensor_readings_col_defs, ...addtional_columnDefs],
```
Comment (nllong) [inline seed/static/seed/js/controllers/sensor_readings_upload_modal_controller.js:122]: are the `...` correct?

File: seed/tests/data/small_sensor_readings_example_bad.csv [hunk 3]
```diff
@@ -0,0 +1,6 @@
+timestamp,I'm not here,my_coolness_sensor,charisma_sensor_1,dex_sensor_1,intelligence_sensor
+2021-10-22 16:30Z,8700000.8,20.7,61,670.9,asdf
+2021-10-22 17:00Z,90.5,21.7,56.7,704.8,207.9
+2021-10-22 17:15Z,90,20.8,59.2,654.1,163.8
+22/12/2021 17:30Z,87.9,20,61.8,636.2,54.9
+2021-10-22 17:45Z,89.5,20.7,60,653.9,69.2
```
Comment (nllong) [inline seed/tests/data/small_sensor_readings_example_bad.csv:6]: add trailing carriage return/line feed/new line

**Review by nllong — APPROVED** [review]
looks good. 

Once we get a more stable input file, then we should add a "download template" button from the sensor/sensorreading upload modals

image

File: seed/static/seed/js/controllers/sensor_readings_upload_modal_controller.js (hunk 2 above)
Comment (haneslinger) [reply seed/static/seed/js/controllers/sensor_readings_upload_modal_controller.js:122]: It's [spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_array_literals) syntax.  🪄

**Comment by RDmitchell** [comment]
@nllong -- is the upload modal the best place to put a button to download the template? Maybe somewhere else as well?

**Comment by nllong** [comment]
yeah, we could just put it on the header of that page too.

**Pull request merged**